### PR TITLE
rbi: add rbi scripts, rbci dockerfile for kata-agent

### PR DESCRIPTION
**Pull request by Xynnn007**
for reproducible buildings of IC components inside HW-TEE, including
- reproducible build container image (RBCI) dockerfile for kata-agent
- patch for kata-agent stable-2.1 to make it reproducible
- scripts for build RBCI, making RB tests for both local and remote source code from github
- readme.md for usage of the scripts

### CODE REVIEW

File: rbi/kata-agent/readme.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# kata-agent的RB
+
+## RB 实现途径
+标准代码是`stable2.1`版本，加上给出的patch，得到的`kata-agent`的sha256值为
```
Comment (jiazhang0) [inline rbi/kata-agent/readme.md:4]: If you provide Chinese version of doc, please rename the file name to readme_cn.md. In addition, the English doc must be provided.

File: rbi/readme.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+```
+上述两个命令，均将会以二进制文件+报告的形式将结果输出到`report`下
+删除RBCI
+```bash
+./rbi.sh agent-image
+```
+清除所有临时文件
+```bash
+./rbi clean
+```
```
Comment (jiazhang0) [inline rbi/readme.md:29]: Move the Chinese content to a dedicated file.

File: rbi/kata-agent/kata-build-docker.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
+#! /bin/bash
```
Comment (jiazhang0) [inline rbi/kata-agent/kata-build-docker.sh:1]: add an empty line.

File: rbi/kata-agent/kata-source-code.sh [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+main() {
+    download_code
+
+    checkout_patch
+}
+
+main "$@"
+
```
Comment (jiazhang0) [inline rbi/kata-agent/kata-source-code.sh:48]: remove these useless lines

File: rbi/kata-agent/kata-test.sh [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    fi
+
+    run_build
+
+    # end_notify
+}
+
+main "$@"
+
+
```
Comment (jiazhang0) [inline rbi/kata-agent/kata-test.sh:73]: ditto

File: rbi/kata-agent/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,17 @@
+# ARG IMAGE_REGISTRY=registry.fedoraproject.org
+
+FROM centos:8.3.2011
+
```
Comment (jiazhang0) [inline rbi/kata-agent/Dockerfile:4]: Please add your maintainer info like `LABEL maintainer="Who <email>"`

File: rbi/kata-agent/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+RUN [ -n "$http_proxy" ] && sed -i '$ a proxy='$http_proxy /etc/dnf/dnf.conf ; true
+RUN dnf install make git gcc -y 
+
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y --no-modify-path
+ENV PATH "/root/.cargo/bin:${PATH}"
+RUN rustup install 1.53.0 && rustup target add x86_64-unknown-linux-musl
+
+VOLUME ["/root/input", "/root/output"]
```
Comment (hustliyilin) [inline rbi/kata-agent/Dockerfile:12]: I think the VOLUME command in the dockerfile isn't always safe. If you really need it, you can use 'docker run -v' to replace it.

**Pull request merged**